lib/fibered_mysql2/fibered_database_connection_pool.rb

```diff
@@ -2,239 +2,35 @@
-#        When we have configurable logging we can put this back and have it off by default.
-
-require 'em-synchrony'
-require 'em-synchrony/thread'
-require 'fibered_mysql2/fibered_mutex_with_waiter_priority'
-
-EventMachine::Synchrony::Thread::Mutex.prepend(FiberedMysql2::FiberedMutexWithWaiterPriority)
-
 module FiberedMysql2
-  class FiberedConditionVariable
-    EXCEPTION_NEVER = {Exception => :never}.freeze
-    EXCEPTION_IMMEDIATE = {Exception => :immediate}.freeze
-
-    #
-    # FIXME: This isn't documented in Nutshell.
-    #
-    # Since MonitorMixin.new_cond returns a ConditionVariable, and the example
-    # above calls while_wait and signal, this class should be documented.
-    #
-    class Timeout < Exception; end
-
-    #
-    # Releases the lock held in the associated monitor and waits; reacquires the lock on wakeup.
-    #
-    # If +timeout+ is given, this method returns after +timeout+ seconds passed,
-    # even if no other thread doesn't signal.
-    #
-    def wait(timeout = nil)
-      Thread.handle_interrupt(EXCEPTION_NEVER) do
-        @monitor.__send__(:mon_check_owner)
-        count = @monitor.__send__(:mon_exit_for_cond)
-        begin
-          Thread.handle_interrupt(EXCEPTION_IMMEDIATE) do
-            @cond.wait(@monitor.instance_variable_get(:@mon_mutex), timeout)
-          end
-          return true
-        ensure
-          @monitor.__send__(:mon_enter_for_cond, count)
-        end
-      end
-    end
-
-    #
-    # Calls wait repeatedly while the given block yields a truthy value.
-    #
-    def wait_while
-      while yield
-        wait
-      end
-    end
-
-    #
-    # Calls wait repeatedly until the given block yields a truthy value.
-    #
-    def wait_until
-      until yield
-        wait
-      end
-    end
-
-    #
-    # Wakes up the first thread in line waiting for this lock.
-    #
-    def signal
-      @monitor.__send__(:mon_check_owner)
-      @cond.signal
-    end
-
-    #
-    # Wakes up all threads waiting for this lock.
-    #
-    def broadcast
-      @monitor.__send__(:mon_check_owner)
-      @cond.broadcast
-    end
-
-    def initialize(monitor)
-      @monitor = monitor
-      @cond = EM::Synchrony::Thread::ConditionVariable.new
-    end
-  end
-
-  # From Ruby's MonitorMixin, with all occurrences of Thread changed to Fiber
-  module FiberedMonitorMixin
-    def self.extend_object(obj)
-      super
-      obj.__send__(:mon_initialize)
-    end
-
-    #
-    # Attempts to enter exclusive section.  Returns +false+ if lock fails.
-    #
-    def mon_try_enter
-      if @mon_owner != Fiber.current
-        @mon_mutex.try_lock or return false
-        @mon_owner = Fiber.current
-        @mon_count = 0
-      end
-      @mon_count += 1
-      true
+  module FiberedDatabaseConnectionPool
+    case ::Rails::VERSION::MAJOR
+    when 6
+    else
+      raise ArgumentError, "unexpected Rails version #{Rails::VERSION::MAJOR}"
```

Comment:
Are we trying to preserve this `case` structure for supporting Rails 7+ in the future? It's a little awkward while Rails 6 is the only supported major version.

Speaking of: can the gemspec be updated to advertise this contract?

Reply:
Good points! Fixed: https://github.com/Invoca/fibered_mysql2/pull/9/commits/87996bc250f0a812f4bd3e9b9ff80245745a1666